base/range.jl

```diff
@@ -58,6 +50,21 @@ function steprange_last{T}(start::T, step, stop)
     last
 end
 
+function steprange_last_empty{T<:Integer}(start::T, step, stop)
+    # empty range has a special representation where stop = start-1
+    # this is needed to avoid the wrap-around that can happen computing
+    # start - step, which leads to a range that looks very large instead
+    # of empty.
+    if step > zero(step)
+        last = start - one(stop-start)
+    else
+        last = start + one(stop-start)
+    end
+    last
+end
+# For types where x+one(x) may not be well-defined
+steprange_last_empty(start, step, stop) = start - step
```

Comment:
Wouldn't this be a more general way of handling overflow issues without using `one` or `zero`?
```julia
function steprange_last_empty(start, step, stop)
	prev = start - step
	last = prev < start ? prev : start + step
	return last
end
```